src/lightning/fabric/utilities/distributed.py

```diff
@@ -300,6 +304,13 @@ def _init_dist_connection(
     )
 
 
+def _destroy_dist_connection() -> None:
+    if _distributed_is_initialized():
+        # ensure at least one collective op ran, otherwise `destroy_process_group()` hangs
+        torch.distributed.barrier()
+        torch.distributed.destroy_process_group()
```

Comment:
Should we check whether the user already destroyed the process group already somehow?

Reply:
You are right, definitely important. It's covered by the `if _distributed_is_initialized():` statement already.